lldb/tools/lldb-dap/lldb-dap.cpp

```diff
@@ -4028,6 +4032,9 @@ void request_variables(DAP &dap, const llvm::json::Object &request) {
             dap.enable_synthetic_child_debugging,
             /*is_name_duplicated=*/false, custom_name));
       };
+      if (variable.GetType().IsPointerType() || variable.GetType().IsReferenceType()) {
+        variable = variable.Dereference();
+      }
```

Comment:
Do you know why you need this change and the one above is not enough?

Reply:
The first change is more than enough. To be on the safe side, I overcomplicated it a little.